flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractTaskClientGenerator.java

```diff
@@ -66,18 +67,37 @@ public void execute() throws ExecutionFailedException {
         File generatedFile = getGeneratedFile();
         try {
             String fileContent = getFileContent();
-            log().debug("writing file '{}'", generatedFile);
-
-            FileUtils.forceMkdirParent(generatedFile);
-            FileUtils.writeStringToFile(generatedFile, fileContent, UTF_8);
+            writeIfChanged(generatedFile, fileContent);
         } catch (IOException exception) {
             String errorMessage = String.format("Error writing '%s'",
                     generatedFile);
             throw new ExecutionFailedException(errorMessage, exception);
         }
     }
 
-    Logger log() {
-        return LoggerFactory.getLogger(getClass());
+    static Logger log() {
+        return LoggerFactory.getLogger(AbstractTaskClientGenerator.class);
+    }
+
+    static void writeIfChanged(File file, String content) throws IOException {
+        String existingFileContent = getExistingFileContent(file);
+        if (content.equals(existingFileContent)) {
+            // Do not write the same contents to avoid frontend recompiles
+            return;
+        }
+
+        log().debug("writing file '{}'", file);
+
+        FileUtils.forceMkdirParent(file);
+        FileUtils.writeStringToFile(file, content, UTF_8);
+    }
+
+    private static String getExistingFileContent(File generatedFile)
+            throws IOException {
+        if (!generatedFile.exists()) {
+            return null;
+        }
+        return FileUtils.readFileToString(generatedFile,
+                StandardCharsets.UTF_8);
```

Comment:
nit: we can maybe use a static import for `UTF_8`, as we do in `writeIfChanged`
```suggestion
        return FileUtils.readFileToString(generatedFile, UTF_8);
```

Reply:
Done